modules/infra/control/iface.c

```diff
@@ -71,14 +71,15 @@ struct iface *iface_create(
 ) {
 	struct iface_type *type = iface_type_get(type_id);
 	struct iface *iface = NULL;
+	const struct iface *i;
 	uint16_t ifid;
 
 	if (type == NULL)
 		goto fail;
 	if (utf8_check(name, GR_IFACE_NAME_SIZE) < 0)
 		goto fail;
-	while ((iface = iface_next(GR_IFACE_TYPE_UNDEF, iface)) != NULL) {
-		if (strcmp(name, iface->name) == 0) {
+	while ((i = iface_next(GR_IFACE_TYPE_UNDEF, i)) != NULL) {
+		if (strcmp(name, i->name) == 0) {
 			errno = EEXIST;
```

Comment:
maybe just set "iface = NULL;" here, instead of defining a new var ?

Reply:
Good point :)

Done.